### PR TITLE
[0.71] Update macOS init template to enable New Architecture

### PR DESCRIPTION
**Pull request by shwanton**
#### Please select one of the following
- [ ] I am removing an existing difference between facebook/react-native and microsoft/react-native-macos :thumbsup:
- [ ] I am cherry-picking a change from Facebook's react-native into microsoft/react-native-macos :thumbsup:
- [x] I am making a fix / change for the macOS implementation of react-native
- [ ] I am making a change required for Microsoft usage of react-native

## Summary

**The default template was never upgraded to 0.71 and was not loading the new arch when `RCT_NEW_ARCH_ENABLED=1` flag was passed to `pod install`

## Changelog

[MACOS] [FIXED] - Update macOS init template to enable New Architecture

## Test Plan

* Edit `packages/react-native-macos-init/src/cli.ts` and remove lines where it clones `react-native-macos`
* Run `yarn build` from `packages/react-native-macos-init/package.json`

* Run `npx react-native-macos` from local fork inside of newly created RN macos app
    * `npx ../shwanton_react-native-macos/packages/react-native-macos-init`

CleanShot 2023-06-16 at 07 36 15
CleanShot 2023-06-16 at 07 35 33

### CODE REVIEW

File: Libraries/AppDelegate/RCTAppDelegate.mm [hunk 1]
```diff
@@ -79,11 +79,19 @@ - (void)applicationDidFinishLaunching:(NSNotification *)notification
...
+                                              backing:NSBackingStoreBuffered
+                                                defer:NO];
+  self.window.title = self.moduleName;
+  self.window.autorecalculatesKeyViewLoop = YES;
+  NSViewController *rootViewController = [NSViewController new];
   rootViewController.view = rootView;
+  rootView.frame = frame;
   self.window.contentViewController = rootViewController;
   [self.window makeKeyAndOrderFront:self];
+  [self.window center];
```
Comment (shwanton) [inline Libraries/AppDelegate/RCTAppDelegate.mm:94]: Taken from how we load RNTester for macOS

https://github.com/microsoft/react-native-macos/blob/04bf38c0c54595b0ac30db790d360719af228f6d/packages/rn-tester/RNTester/AppDelegate.mm#L129-L143

**Review by Saadnajmi — APPROVED** [review]

File: local-cli/generate-macos.js [hunk 2]
```diff
@@ -30,7 +30,7 @@ function generateMacOS (projectDir, name, options) {
     { overwrite: options.overwrite }
   );
 
-  installPods(options);
+  // installPods(options);
```
Comment (shwanton) [inline local-cli/generate-macos.js:33]: oops, will uncomment

**Comment by Saadnajmi** [comment]
Can you bring to main too?

File: local-cli/generator-macos/templates/macos/Podfile [hunk 3]
```diff
@@ -1,32 +1,28 @@
...
-  # Enables Flipper.
-  #
-  # Note that if you have use_frameworks! enabled, Flipper will not work and
-  # you should disable these next few lines.
-  use_flipper!
-
   post_install do |installer|
     react_native_post_install(installer)
     __apply_Xcode_12_5_M1_post_install_workaround(installer)
   end
```
Comment (shwanton) [inline local-cli/generator-macos/templates/macos/Podfile:27]: Still needed for macOS target to fix this issue:
https://github.com/facebook/react-native/issues/32483

Comment (Saadnajmi) [reply local-cli/generator-macos/templates/macos/Podfile:27]: Bumping folly in React Native upstream is hard :')

**Comment by shwanton** [comment]
closes #1841 #1821

**Pull request merged**